lib/ansible/playbook/role/__init__.py

```diff
@@ -376,7 +376,10 @@ def serialize(self, include_deps=True):
 
         parents = []
         for parent in self._parents:
-            parents.append(parent.serialize(include_deps=False))
+            # include parent only once
+            # Fix https://github.com/ansible/ansible/issues/14112
+            if parent not in self._parents:
```

Comment:
@jimi-c Well, you're right ... This condition is always false so it's quite normal to get a huge speed improvement.